app/modules/users/permissions/rules.py

```diff
@@ -16,6 +16,14 @@
 log = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+PERMISSION_ROLE_MAP = {
+    ('SiteSetting', AccessOperation.READ, 'Module'): ['is_privileged', 'is_admin'],
```

Comment:
@hwindsor What do you think about rewriting the permissions code more like this?

Reply:
I like it.

Reply:
Yeah I'm liking the structure, will definitely be cleaner than what is currently growing.
The only concern is that maybe user role driven permissions and object public driven permissions should be in different lists. 
The commented out project and org code will need to be restored at some point too, also collaborations. I'm thinking that they should be different maps but generally yes, I like it.

Reply:
I'm going to merge this in. If we need to further polish the code it can be done in a subsequent PR.